pkg/controllers/plugin/util.go

```diff
@@ -164,27 +172,41 @@ func isPayloadReadyRunning(o interface{}) bool {
 	return false
 }
 
-func computeReadyCondition(pluginStatus greenhousev1alpha1.PluginStatus, release *ReleaseStatus) greenhousev1alpha1.Condition {
+// allResourceReady checks if all resources are ready
+func allResourceReady(payloadStatus []PayloadStatus) bool {
+	for _, status := range payloadStatus {
+		if !status.Ready {
+			return false
+		}
+	}
+	return true
+}
+
+// computeReadyCondition computes the ReadyCondition for the Plugin and sets the workload metrics and message
+func computeReadyCondition(plugin *greenhousev1alpha1.Plugin, pluginStatus greenhousev1alpha1.PluginStatus, release *ReleaseStatus) greenhousev1alpha1.Condition {
 	WorkloadReadyStatus := *pluginStatus.GetConditionByType(greenhousev1alpha1.WorkloadReadyCondition)
 
-	if release.ReleaseOK {
-		WorkloadReadyStatus.Status = metav1.ConditionTrue
-		WorkloadReadyStatus.Message = "Workload is running"
-	} else {
+	WorkloadReadyStatus.Status = metav1.ConditionTrue
+	if !allResourceReady(release.PayloadStatus) {
 		WorkloadReadyStatus.Status = metav1.ConditionFalse
-		WorkloadReadyStatus.Message = release.Message
+		setWorkloadMetrics(plugin, 0)
+		WorkloadReadyStatus.Message = "Following workload resources are not ready: [ "
+		for _, status := range release.PayloadStatus {
+			if !status.Ready {
+				WorkloadReadyStatus.Message += status.Message + ", "
+			}
+		}
+		WorkloadReadyStatus.Message += " ]"
+	} else {
+		setWorkloadMetrics(plugin, 1)
+		WorkloadReadyStatus.Message = "Workload is running"
 	}
+
 	return WorkloadReadyStatus
 }
 
-func logPluginStatus(ctx context.Context, releaseStatus *ReleaseStatus) {
-	marshaled, err := json.Marshal(releaseStatus)
-	if err != nil {
-		log.FromContext(ctx).Info("releaseStatus marshaling error", "error", err)
-	}
-	if releaseStatus.ReleaseOK {
-		log.FromContext(ctx).Info("release is running", "pluginName", releaseStatus.ReleaseName, "Details", string(marshaled))
-	} else {
-		log.FromContext(ctx).Info("release is not running", "pluginName", releaseStatus.ReleaseName, "Details", string(marshaled))
-	}
+// setWorkloadMetrics sets the workload status metric to the given status
+func setWorkloadMetrics(plugin *greenhousev1alpha1.Plugin, status float64) {
+	workloadStatus.WithLabelValues(plugin.GetNamespace(), plugin.Name, plugin.Spec.PluginDefinition).Set(status)
+	workloadStatus.WithLabelValues(plugin.GetNamespace(), plugin.Name, plugin.Spec.PluginDefinition).SetToCurrentTime()
```

Comment:
This will set the metric value to the current timestamp, but we want to have 1 or 0 to indicate if currently the resources are up or not.
```suggestion
	workloadStatus.WithLabelValues(plugin.GetNamespace(), plugin.Name, plugin.Spec.PluginDefinition).Set(status)
```